src/components/CentralContainer/Sidebar/SidebarTabs/SearchTabPanel/index.tsx

```diff
@@ -0,0 +1,119 @@
+import TitleButton from "../TitleButton";
+import ResultsGroup from "./ResultsGroup";
+
+
+enum SEARCH_OPTION {
+    IS_CASE_SENSITIVE = "isCaseSensitive",
+    IS_REGEX = "isRegex"
+}
+
+/**
+ * Displays a panel for submitting search queries and viewing query results.
+ *
+ * @return
+ */
+const SearchTabPanel = () => {
+    const [isAllExpanded, setIsAllExpanded] = useState<boolean>(true);
+    const [searchOptions, setSearchOptions] = useState<SEARCH_OPTION[]>([]);
+    const searchTextRef = useRef<HTMLTextAreaElement>(null);
+    const {queryResults, startQuery} = useContext(StateContext);
+    const handleSearch = (event: React.KeyboardEvent<HTMLTextAreaElement>) => {
+        if ("Enter" === event.key && searchTextRef.current) {
+            event.preventDefault();
+            const isCaseSensitive = searchOptions.includes(SEARCH_OPTION.IS_CASE_SENSITIVE);
+            const isRegex = searchOptions.includes(SEARCH_OPTION.IS_REGEX);
+            startQuery(searchTextRef.current.value, isRegex, isCaseSensitive);
+        }
+    };
+    const {queryProgress} = useContext(StateContext);
+
+    return (
+        <CustomTabPanel
+            tabName={TAB_NAME.SEARCH}
+            title={TAB_DISPLAY_NAMES[TAB_NAME.SEARCH]}
+            titleButtons={
+                <TitleButton onClick={() => { setIsAllExpanded((v) => !v); }}>
+                    {isAllExpanded ?
+                        <UnfoldLessIcon/> :
+                        <UnfoldMoreIcon/>}
+                </TitleButton>
+            }
+        >
+            <Textarea
+                maxRows={7}
+                placeholder={"Search"}
+                size={"sm"}
+                sx={{flexDirection: "row", zIndex: 0}}
+                endDecorator={
+                    <ToggleButtonGroup
+                        size={"sm"}
+                        spacing={0.3}
+                        sx={{borderRadius: "2px"}}
+                        value={searchOptions}
+                        variant={"plain"}
+                        onChange={(_, newValue) => {
+                            setSearchOptions(newValue);
+                        }}
+                    >
+                        <IconButton
+                            sx={{fontFamily: "Inter"}}
+                            value={SEARCH_OPTION.IS_CASE_SENSITIVE}
+                        >
+                            Aa
+                        </IconButton>
+                        <IconButton
+                            sx={{fontFamily: "Inter"}}
+                            value={SEARCH_OPTION.IS_REGEX}
+                        >
+                            .*
+                        </IconButton>
+                    </ToggleButtonGroup>
+                }
+                slotProps={{
+                    textarea: {ref: searchTextRef},
+                    endDecorator: {sx: {marginBlockStart: 0, display: "block"}},
+                }}
+                onKeyDown={handleSearch}/>
+            <LinearProgress
```

Comment:
The progress bar does seem a bit too thick. Can we tune its `thickness={2}` or {1}`?